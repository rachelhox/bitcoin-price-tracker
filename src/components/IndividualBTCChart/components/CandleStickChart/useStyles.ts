import { css, useTheme } from '@mui/material';

const useStyles = () => {
  const theme = useTheme();
  return {
    root: css({
      margin: 'auto',
      [theme.breakpoints.up('laptop')]: {
        width: '70%',
      },
    }),
  };
};

export default useStyles;
