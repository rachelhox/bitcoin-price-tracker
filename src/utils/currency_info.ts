interface IStringIndex extends Record<string, any> {}

/**
 * Takes the key name and returns the location. Will return null if undefined
 */

const currencies: IStringIndex = {
  usd: {
    name: 'US Dollar',
    key: 'usd',
    unit: '$',
    type: 'fiat',
  },
  aed: {
    name: 'United Arab Emirates Dirham',
    key: 'aed',
    unit: 'DH',
    type: 'fiat',
  },
  eur: {
    name: 'Euro',
    key: 'eur',
    unit: '€',
    type: 'fiat',
  },
  aud: {
    name: 'Australian Dollar',
    key: 'aud',
    unit: 'A$',
    type: 'fiat',
  },
  cad: {
    name: 'Canadian Dollar',
    key: 'cad',
    unit: 'CA$',
    type: 'fiat',
  },
  chf: {
    name: 'Swiss Franc',
    key: 'chf',
    unit: 'Fr.',
    type: 'fiat',
  },
  gbp: {
    name: 'British Pound Sterling',
    key: 'gbp',
    unit: '£',
    type: 'fiat',
  },
  hkd: {
    name: 'Hong Kong Dollar',
    key: 'hkd',
    unit: 'HK$',
    type: 'fiat',
  },
  sgd: {
    name: 'Singapore Dollar',
    key: 'sgd',
    unit: 'S$',
    type: 'fiat',
  },
  cny: {
    name: 'Chinese Yuan',
    key: 'cny',
    unit: '¥',
    type: 'fiat',
  },
  jpy: {
    name: 'Japanese Yen',
    key: 'jpy',
    unit: '¥',
    type: 'fiat',
  },
  inr: {
    name: 'Indian Rupee',
    key: 'inr',
    unit: '₹',
    type: 'fiat',
  },
  krw: {
    name: 'South Korean Won',
    key: 'krw',
    unit: '₩',
    type: 'fiat',
  },
};

export const getCurrencyInfo = (key: string | number) => {
  return currencies[key] ?? {};
};
