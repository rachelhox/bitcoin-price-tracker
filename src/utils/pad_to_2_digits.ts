/* eslint-disable */

export const padTo2Digits = (num: any) => {
  return String(num).padStart(2, '0');
};
