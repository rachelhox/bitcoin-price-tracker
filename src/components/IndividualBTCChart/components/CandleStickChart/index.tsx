/* eslint-disable no-unused-vars */
import { Box, useTheme, useMediaQuery } from '@mui/material';
import {
  VictoryChart,
  VictoryAxis,
  VictoryCandlestick,
  VictoryTheme,
  VictoryZoomContainer,
  VictoryTooltip,
} from 'victory';
import { useGeneralContext } from '@contexts';
import { convertToMoney, padTo2Digits } from '@utils';
import { NoSSR } from '@components';
import useStyles from './useStyles';

interface CandleStickChartProp {
  // eslint-disable-next-line no-undef
  candleChartData: OHLCProps[];
  period: number;
}

const CandleStickChart = ({
  candleChartData,
  period,
}: CandleStickChartProp) => {
  const styles = useStyles();
  const theme = useTheme();
  const { currency } = useGeneralContext();
  const onlyLargeScreen = useMediaQuery(theme.breakpoints.up('laptop'));
  return (
    <NoSSR>
      <Box css={styles.root}>
        <VictoryChart
          theme={VictoryTheme.material}
          domainPadding={{ x: 10 }}
          padding={{ left: 90, top: 10, right: 0, bottom: 30 }}
          scale={{ x: 'time' }}
          containerComponent={<VictoryZoomContainer />}
        >
          <VictoryAxis tickFormat={(t) => `${t.getDate()}/${t.getMonth()}`} />
          <VictoryAxis
            dependentAxis
            tickFormat={(t) => [`${currency.unit}${convertToMoney(t)}`]}
          />
          <VictoryCandlestick
            candleColors={{
              positive: theme.palette.custom.green,
              negative: theme.palette.custom.red,
            }}
            candleWidth={5}
            data={candleChartData}
            labels={({ datum }: any) => {
              if (period === 90)
                return [
                  `date: ${datum.x.getDate()}/${datum.x.getMonth()}`,
                  `open: ${currency.unit}${convertToMoney(datum.open)}`,
                  `high: ${currency.unit}${convertToMoney(datum.high)}`,
                  `low: ${currency.unit}${convertToMoney(datum.low)}`,
                  `close: ${currency.unit}${convertToMoney(datum.close)}`,
                ];
              return [
                `time: ${padTo2Digits(datum.x.getHours())}:${padTo2Digits(
                  datum.x.getMinutes()
                )}`,
                `open: ${currency.unit}${convertToMoney(datum.open)}`,
                `high: ${currency.unit}${convertToMoney(datum.high)}`,
                `low: ${currency.unit}${convertToMoney(datum.low)}`,
                `close: ${currency.unit}${convertToMoney(datum.close)}`,
              ];
            }}
            labelComponent={
              <VictoryTooltip
                cornerRadius={8}
                flyoutStyle={{
                  fill: theme.palette.common.black,
                  color: theme.palette.common.white,
                  stroke: theme.palette.common.white,
                }}
                centerOffset={
                  onlyLargeScreen
                    ? {
                        y: undefined,
                        x: undefined,
                      }
                    : {
                        y: -50,
                        x: -50,
                      }
                }
                style={{
                  fill: theme.palette.common.white,
                  fontSize: 8,
                }}
              />
            }
          />
        </VictoryChart>
      </Box>
    </NoSSR>
  );
};

export default CandleStickChart;
