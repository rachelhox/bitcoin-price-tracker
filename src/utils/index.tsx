import { getCurrencyInfo } from './currency_info';
import { convertToMoney } from './convert_to_money';
import { padTo2Digits } from './pad_to_2_digits';

export { getCurrencyInfo, convertToMoney, padTo2Digits };
