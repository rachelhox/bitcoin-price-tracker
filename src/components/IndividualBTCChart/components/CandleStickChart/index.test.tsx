import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import CandleStickChart from '.';

// ==================================
// mocks
// ==================================
const mockCandleChartData = [
  {
    close: 19198.11,
    high: 19383.94,
    low: 19058.63,
    open: 19058.63,
    x: new Date(1667966400000),
  },
];

jest.mock('@components', () => ({
  NoSSR: (props: any) => <div id="NoSSR" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('component: IndividualBTCChart', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <CandleStickChart candleChartData={mockCandleChartData} period={90} />
      </MockTheme>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
