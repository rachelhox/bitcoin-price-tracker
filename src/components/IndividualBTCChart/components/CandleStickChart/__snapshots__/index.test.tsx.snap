// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: IndividualBTCChart matches snapshot 1`] = `
<div
  id="NoSSR"
>
  <div
    className="MuiBox-root css-jtz3ax-CandleStickChart"
  >
    <div
      className="VictoryContainer"
      style={
        {
          "height": "100%",
          "pointerEvents": "none",
          "position": "relative",
          "touchAction": "none",
          "userSelect": "none",
          "width": "100%",
        }
      }
    >
      <svg
        height={350}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onWheel={[Function]}
        role="img"
        style={
          {
            "height": "100%",
            "pointerEvents": "all",
            "width": "100%",
          }
        }
        viewBox="0 0 350 350"
        width={350}
      >
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              {
                "fill": "transparent",
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 2,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={90}
            x2={350}
            y1={320}
            y2={320}
          />
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={90}
              y1={320}
              y2={10}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={90}
              y1={320}
              y2={325}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-0-tickLabels-0"
              x={90}
              y={343.26}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="middle"
                x={90}
              >
                9/10
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={176.66666666666666}
              x2={176.66666666666666}
              y1={320}
              y2={10}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={176.66666666666666}
              x2={176.66666666666666}
              y1={320}
              y2={325}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-0-tickLabels-1"
              x={176.66666666666666}
              y={343.26}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="middle"
                x={176.66666666666666}
              >
                9/10
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={263.3333333333333}
              x2={263.3333333333333}
              y1={320}
              y2={10}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={263.3333333333333}
              x2={263.3333333333333}
              y1={320}
              y2={325}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-0-tickLabels-2"
              x={263.3333333333333}
              y={343.26}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="middle"
                x={263.3333333333333}
              >
                9/10
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={350}
              x2={350}
              y1={320}
              y2={10}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={350}
              x2={350}
              y1={320}
              y2={325}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-0-tickLabels-3"
              x={350}
              y={343.26}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="middle"
                x={350}
              >
                9/10
              </tspan>
            </text>
          </g>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              {
                "fill": "transparent",
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 2,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={90}
            x2={90}
            y1={10}
            y2={320}
          />
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={280.57698810365565}
              y2={280.57698810365565}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={280.57698810365565}
              y2={280.57698810365565}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-0"
              x={77}
              y={284.83698810365564}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,100
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={232.9301281854235}
              y2={232.9301281854235}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={232.9301281854235}
              y2={232.9301281854235}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-1"
              x={77}
              y={237.1901281854235}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,150
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={185.2832682671913}
              y2={185.2832682671913}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={185.2832682671913}
              y2={185.2832682671913}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-2"
              x={77}
              y={189.54326826719128}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,200
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={137.6364083489591}
              y2={137.6364083489591}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={137.6364083489591}
              y2={137.6364083489591}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-3"
              x={77}
              y={141.89640834895908}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,250
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={89.98954843072694}
              y2={89.98954843072694}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={89.98954843072694}
              y2={89.98954843072694}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-4"
              x={77}
              y={94.24954843072695}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,300
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "none",
                  "pointerEvents": "painted",
                  "stroke": "#ECEFF1",
                  "strokeDasharray": "10, 5",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={350}
              y1={42.34268851249477}
              y2={42.34268851249477}
            />
            <line
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "transparent",
                  "size": 5,
                  "stroke": "#90A4AE",
                  "strokeLinecap": "round",
                  "strokeLinejoin": "round",
                  "strokeWidth": 1,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={90}
              x2={85}
              y1={42.34268851249477}
              y2={42.34268851249477}
            />
            <text
              direction="inherit"
              dx={0}
              id="chart-axis-1-tickLabels-5"
              x={77}
              y={46.602688512494765}
            >
              <tspan
                dx={0}
                dy={0}
                style={
                  {
                    "fill": "#455A64",
                    "fontFamily": "'Helvetica Neue', 'Helvetica', sans-serif",
                    "fontSize": 12,
                    "letterSpacing": "normal",
                    "padding": 8,
                    "stroke": "transparent",
                    "strokeWidth": 0,
                  }
                }
                textAnchor="end"
                x={77}
              >
                $19,350
              </tspan>
            </text>
          </g>
        </g>
        <g
          clipPath="url(#victory-clip-2)"
          style={
            {
              "height": "100%",
              "userSelect": "none",
              "width": "100%",
            }
          }
        >
          <defs>
            <clipPath
              id="victory-clip-2"
            >
              <rect
                height={310}
                vectorEffect="non-scaling-stroke"
                width={260}
                x={90}
                y={10}
              />
            </clipPath>
          </defs>
          <g>
            <rect
              height={132.91568042790007}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "#0ECB81",
                  "stroke": "#0ECB81",
                  "strokeWidth": 2,
                }
              }
              vectorEffect="non-scaling-stroke"
              width={5}
              x={260.8333333333333}
              y={187.08431957209993}
            />
            <line
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "#0ECB81",
                  "stroke": "#0ECB81",
                  "strokeWidth": 2,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={263.3333333333333}
              x2={263.3333333333333}
              y1={10}
              y2={187.08431957209993}
            />
            <line
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="presentation"
              shapeRendering="auto"
              style={
                {
                  "fill": "#0ECB81",
                  "stroke": "#0ECB81",
                  "strokeWidth": 2,
                }
              }
              vectorEffect="non-scaling-stroke"
              x1={263.3333333333333}
              x2={263.3333333333333}
              y1={320}
              y2={320}
            />
          </g>
        </g>
      </svg>
      <div
        style={
          {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
            "zIndex": 99,
          }
        }
      >
        <svg
          height={350}
          style={
            {
              "height": "100%",
              "overflow": "visible",
              "width": "100%",
            }
          }
          viewBox="0 0 350 350"
          width={350}
        />
      </div>
    </div>
  </div>
</div>
`;
